### PR TITLE
Call message_post for each record separately

### PR DESCRIPTION
**Pull request by igallart**
Description of the issue/feature this PR addresses:
When two or more record, it fail
Current behavior before PR:
Fail, exception.
Desired behavior after PR is merged:
Loop for each reacord.

https://github.com/OCA/OCB/pull/574

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mart-e** [comment]
Hello,

Thank you for you patch but it seems you have not [signed the CLA](https://www.odoo.com/sign-cla).
Could you add it so we can integrate your patch?
Regards

**Comment by mart-e** [comment]
Hello,

Thank you for the CLA signature but it seems there is still an issue.

- you are signing using the address `igallart@users.noreply.github.com`
- you are signing for the address `i...@puntsistemes.es`

The email address in the CLA must be the same as for the signature for the verification to succeed.
You can either change the your [git configuration](https://help.github.com/articles/setting-your-email-in-git/) or modify the email in the CLA signature.

Thanks

**Comment by igallart** [comment]
Hello, with the path is integred cla? It is wrong?
Thanks


El 24/02/2017 17:24, "Martin Trigaux" <notifications@github.com> escribió:

Hello,

Thank you for you patch but it seems you have not signed the CLA
<https://www.odoo.com/sign-cla>.
Could you add it so we can integrate your patch?
Regards

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/odoo/odoo/pull/15625#issuecomment-282334607>, or mute
the thread

**Comment by mart-e** [comment]
> Hello, with the commit is integred cla. It is wrong?

Please discard my first email, went too fast in the reply. The issue comes from the fact that email address making the commit is not the same as the one you are signing for.

File: addons/mrp/mrp.py [hunk 1]
```diff
@@ -783,9 +783,9 @@ def action_cancel(self, cr, uid, ids, context=None):
         # Put related procurements in exception
         proc_obj = self.pool.get("procurement.order")
         procs = proc_obj.search(cr, uid, [('production_id', 'in', ids)], context=context)
-        if procs:
-            proc_obj.message_post(cr, uid, procs, body=_('Manufacturing order cancelled.'), context=context)
-            proc_obj.write(cr, uid, procs, {'state': 'exception'}, context=context)
+        for proc in procs:
+            proc_obj.message_post(cr, uid, proc, body=_('Manufacturing order cancelled.'), context=context)
+            proc_obj.write(cr, uid, proc, {'state': 'exception'}, context=context)
```
Comment (mart-e) [inline addons/mrp/mrp.py]: The error is with the `message_post` taking only one id, right?
The write can be set outside of the loop for better performance

**Comment by igallart** [comment]
Sorry @mart-e, currently I can not reproduce it, was in 2016 december (https://github.com/OCA/OCB/pull/574), and I not remember, but the was fixed with the commit, and was related with messenger in that lines. Sorry by email, I have now changed config in git.

File: addons/mrp/mrp.py (hunk 1 above)
Comment (igallart) [inline addons/mrp/mrp.py]: Yes, that is the error, fail message_post

**Comment by mart-e** [comment]
> Sorry by email, I have now changed config in git.

As said, you need to make the commit with the same address. You may have changed your local configuration but this has no impact on the pull request. Could you [amend the two commits](http://stackoverflow.com/a/1320317/147904) to change the author and push force is this branch?

Also, you can move the write outside of the loop.
Thanks

**Comment by igallart** [comment]
I think that is correct!

**Comment by mart-e** [comment]
Thanks for the update, merged at 48cebafdafd

**Pull request closed without merging**